src/map/Map.js

```diff
@@ -452,10 +457,15 @@ L.Map = L.Evented.extend({
 	// @method setMaxZoom(zoom: Number): this
 	// Sets the upper limit for the available zoom levels (see the [maxZoom](#map-maxzoom) option).
 	setMaxZoom: function (zoom) {
+		var oldZoom = this.options.maxZoom;
 		this.options.maxZoom = zoom;
 
-		if (this._loaded && (this.getZoom() > this.options.maxZoom)) {
-			return this.setZoom(zoom);
+		if (this._loaded && oldZoom !== zoom) {
+			if (this.getZoom() > this.options.maxZoom) {
+				return this.setZoom(zoom);
+			} else {
+				this.fire('zoomlevelschange');

```

Comment:
Same here, it looks like the event should fire even if we have to call `setZoom`, no?